experimental/oak_baremetal_simple_io/src/lib.rs

```diff
@@ -53,6 +59,10 @@ pub const INPUT_BUFFER_LEGNTH: usize = 4096;
 /// A Simple IO device implementation that uses direct port-based IO.
 pub type RawSimpleIo<'a> = SimpleIo<'a, PortReadOnly<u32>, PortWriteOnly<u32>>;
 
+/// Memory address translation function.
+pub trait Translator: Fn(VirtAddr) -> Option<PhysAddr> {}
```

Comment:
Alternatively this may just be a type alias, then I think you can avoid the blanket impl below.

Reply:
That was my first attempt, it didn't work.

Reply:
The trick I used here is from this post: https://users.rust-lang.org/t/why-cant-type-aliases-be-used-for-traits/10002